packages/core/test/new-api/new-execution/reducers/deployment-state-reducer.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  DeploymentExecutionStateInitializeMessage,
+  JournalMessage,
+  JournalMessageType,
+  NetworkInteractionRequestMessage,
+  SendTransactionMessage,
+} from "../../../../src/new-api/internal/new-execution/types/messages";
+import { NetworkInteractionType } from "../../../../src/new-api/internal/new-execution/types/network-interaction";
+import { assertIgnitionInvariant } from "../../../../src/new-api/internal/utils/assertions";
+import { FutureType } from "../../../../src/new-api/types/module";
+
+describe("DeploymentStateReducer", () => {
+  const exampleAddress = "0x1F98431c8aD98523631AE4a59f267346ea31F984";
+
+  let initialState: DeploymentState;
+  let updatedState: DeploymentState;
+
+  describe("starting a new run", () => {
+    beforeEach(() => {
+      initialState = deploymentStateReducer(undefined);
+
+      updatedState = deploymentStateReducer(initialState, {
+        type: JournalMessageType.RUN_START,
+        chainId: 31337,
+      });
+    });
+
+    it("should set the chainId", () => {
+      assert.equal(updatedState.chainId, 31337);
+    });
+
+    it("should leave the previous execution states", () => {
+      assert.equal(initialState.executionStates, updatedState.executionStates);
+    });
+  });
+
+  describe("running a named contract deploy", () => {
+    const initializeNamedContractDeployMessage: DeploymentExecutionStateInitializeMessage =
+      {
+        type: JournalMessageType.DEPLOYMENT_EXECUTION_STATE_INITIALIZE,
+        futureId: "future1",
+        futureType: FutureType.NAMED_CONTRACT_DEPLOYMENT,
+        strategy: "basic",
+        dependencies: [],
+        artifactFutureId: "future1",
+        contractName: "MyContract",
+        constructorArgs: [],
+        libraries: {},
+        value: BigInt(0),
+        from: undefined,
+      };
+
+    const requestNetworkInteractionMessage: NetworkInteractionRequestMessage = {
+      type: JournalMessageType.NETWORK_INTERACTION_REQUEST,
+      futureId: "future1",
+      networkInteraction: {
+        id: 1,
+        type: NetworkInteractionType.ONCHAIN_INTERACTION,
+        to: exampleAddress,
+        data: "fake-data",
+        value: BigInt(0),
+        from: "string",
+        nonce: 0,
+        transactions: [],
+      },
+    };
+
+    const sendTransactionMessage: SendTransactionMessage = {
+      type: JournalMessageType.TRANSACTION_SEND,
+      futureId: "future1",
+      networkInteractionId: 1,
+      transaction: {
+        hash: "0xdeadbeef",
+        maxFeePerGas: BigInt(10),
+        maxPriorityFeePerGas: BigInt(5),
+      },
+    };
+
+    describe("initialization", () => {
+      beforeEach(() => {
+        updatedState = applyMessages([initializeNamedContractDeployMessage]);
+      });
+
+      it("should populate a deployment execution state for the future", () => {
+        const exState = updatedState.executionStates.future1;
+        assert.isDefined(exState);
+        assert(isDeploymentExecutionState(exState));
```

Comment:
This could be `assert.equal(exState.type, ...)`, so that we don't need a type guard, and also the error message would be better if it fails.